src/java.base/share/classes/java/lang/reflect/Proxy.java

```diff
@@ -838,12 +823,13 @@ private static Module mapToModule(ClassLoader loader,
                     Modules.addOpens(targetModule, targetPackageName, Proxy.class.getModule());
                 }
                 // return the module of the package-private interface
-                return targetModule;
+                return new ProxyContext(targetModule, targetPackageName, true);
             }
 
             // All proxy interfaces are public.  So maps to a dynamic proxy module
             // and add reads edge and qualified exports, if necessary
-            Module targetModule = getDynamicModule(loader);
+            var context = getDynamicModuleContext(loader, nonExported);
```

Comment:
I suggest to keep the `getDynamicModule` method as creating a dynamic module of a given class loader is a clear function.   The context can be created in this method body.

```
            var targetModule = getDynamicModule(loader);
            var pkgName = nonExported ? PROXY_PACKAGE_PREFIX + '.' + targetModule.getName()
                                      : targetModule.getName();
            var context = new ProxyClassContext(targetModule, pkgName, Modifier.PUBLIC);
```